### PR TITLE
Try out depfiles for nightly testing

### PR DESCRIPTION
**Pull request by psakievich**
@jrood-nrel @tasmith4 something we could test out one night in the future. This is an embarrassingly parallel build process I tried to day on a regular environment. 

https://spack.readthedocs.io/en/latest/environments.html#generating-depfiles-from-environments

### CODE REVIEW

**Comment by tasmith4** [comment]
@psakievich looks like it is missing the actual depfile?

**Comment by psakievich** [comment]
It isn't. Makefile is the depfile. It's pretty cool. When you call `make` it DAG and package level parallelisms are combined.

**Comment by tasmith4** [comment]
ok yeah I don't see a problem trying this sometime.  whenever it works for @jrood-nrel is fine with me, as far as I'm concerned if it breaks our nightlies too bad we can always revert

File: scripts/run-exawind-nightly-tests.sh [hunk 1]
```diff
@@ -132,10 +132,13 @@ cmd "spack concretize -f --fresh"
 
 printf "\nTests started at: $(date)\n\n"
 printf "spack install \n"
+spack env depfile -o Makefile
 if [ "${SPACK_MANAGER_MACHINE}" == 'ascicgpu' ]; then
-  time (spack install --keep-stage)
+  # time (spack install --keep-stage)
+  time (make -j32)
```
Comment (psakievich) [inline scripts/run-exawind-nightly-tests.sh]:
```suggestion
  time (make SPACK_INSTALL_FLAGS=--keep-stage -j32)
```

File: scripts/run-exawind-nightly-tests.sh [hunk 2]
```diff
@@ -132,10 +132,13 @@ cmd "spack concretize -f --fresh"
...
 else
-  time (for i in {1..4}; do spack install --keep-stage --fresh & done; wait)
+  # time (for i in {1..4}; do spack install --keep-stage --fresh & done; wait)
+  time (make -j32)
```
Comment (psakievich) [inline scripts/run-exawind-nightly-tests.sh]:
```suggestion
  time (make SPACK_INSTALL_FLAGS='--keep-stage --fresh' -j32)
```

**Review by tasmith4 — APPROVED** [review]

**Comment by jrood-nrel** [comment]
I'm fine with this if it's a `make -j4`.

**Comment by psakievich** [comment]
> I'm fine with this if it's a `make -j4`.

Hmm good point. We should probably test this out on the matrix to see how it does before merging.  Let's just leave it open for a bit until we have time to test it out

**Comment by psakievich** [comment]
> I'm fine with this if it's a `make -j4`.

@jrood-nrel is this out of concern that we'd be running too many concurrent tests on the machines? 4 cores seems really small.

**Comment by jrood-nrel** [comment]
Sometimes Nalu is running 16 processes (build_jobs=16 on rhodes) at a time and they look like they can be 2GB per process. If we have more than 4x of those concurrently, I think it could be a problem for the test machine that has 128GB of RAM.

**Comment by psakievich** [comment]
It seems like we should split the dependency build from the tests before doing this. We can unleash the full build on the dependencies and then do a linear spack install for the tests so they run 1 at a time?

**Comment by jrood-nrel** [comment]
I would still like to run the test categories in parallel. The clang tests take 12 hours.

**Pull request closed without merging**